### PR TITLE
fix: avoid redaction of response headers

### PR DESCRIPTION
**Pull request by david-luna**
# Description

This PR intends to finish the work started in https://github.com/elastic/apm-agent-nodejs/pull/3382 which has a fix to nor redact response headers. This one includes the fix on `redactObjectKeys` from the aforementioned PR and also changes the usage of it in `parser.js` file.

### Checklist

- [x] Implement code
- [x] Add tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -41,6 +41,8 @@ Notes:
 
 * Add support for 'knex' version v1 and v2. ({pull}3355[#3355])
 * Add `tedious@16.x` support. ({pull}3366[#3366])
+* Add `apm.setGlobalLabel()` to dynamically extend the `globalLabels` set in the initial config.
```
Comment (david-luna) [inline CHANGELOG.asciidoc:44]: this line was misplaced. the feature will be available in next release

File: lib/parsers.js [hunk 2]
```diff
@@ -102,6 +99,25 @@ function getContextFromRequest (req, conf, type) {
...
+ *    Some relevant fields: (TODO: document all used fields)
+ *    - `headers` - Required. An object.
+ *    - `statusCode` - Required. A number.
+ *    - `headersSent` - Boolean indicating if the headers have been sent
+ *       (https://nodejs.org/api/http.html#outgoingmessageheaderssent)
+ *    - `finished` - Boolean indicating if `response.end()` has been called
+ *       (https://nodejs.org/api/http.html#responsefinished)
+ * @param {Object} conf - The full agent configuration.
+ * @param {Boolean} isError - Indicates if this response contains an error and
+ *    some extar fields should be added to the context
```
Comment (trentm) [inline lib/parsers.js:119]:
```suggestion
 *    some extra fields should be added to the context
```

typo

File: lib/parsers.js [hunk 3]
```diff
@@ -102,6 +99,25 @@ function getContextFromRequest (req, conf, type) {
...
+/**
+ * Extract appropriate `{transaction,error}.context.response` from an HTTP
+ * response object. This handles header redaction according to the agent config.
+ *
+ * @param {Object} res - Typically `res` is a Node.js `http.OutgoingMessage`
+ *    (https://nodejs.org/api/http.html#class-httpoutgoingmessage).
+ *    However, some cases (e.g. Lambda and Azure Functions instrumentation)
+ *    create a pseudo-res object that matches well enough for this function.
+ *    Some relevant fields: (TODO: document all used fields)
+ *    - `headers` - Required. An object.
```
Comment (trentm) [inline lib/parsers.js:111]: *Is* `res.headers` now required? Note that there is a code path in lambda.js that sets a pseudo `trans.res` that does *not* have a `headers` field: https://github.com/elastic/apm-agent-nodejs/blob/9f13472d69523109d69315c6bb212957e46809cb/lib/lambda.js#L225-L228

What happens in that case?

Comment (trentm) [reply lib/parsers.js]: I added commit d4c5aae2007b2586cba5a90b322eaf7cd6638266 that shows the headers aren't *required*, which sanity checks that this won't crash in that case.

**Comment by trentm** [comment]
@david-luna and I discussed offline the other day that this *could* result in creating more objects that absolutely necessary. However, we don't expect it to be a real/significant perf issue.

**Comment by trentm** [comment]
Merge from main was to get https://github.com/elastic/apm-agent-nodejs/pull/3431 to fix a recently-flaky  test in http-request.test.js.

**Review by trentm — APPROVED** [review]

**Pull request merged**